From 889835cd5c56e8871f01de1e1e8d038343648dfd Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 27 Sep 2024 08:41:20 +0200
Subject: [PATCH 1312/1695] drivers: stm32_cpu_opp: fix against GCC 14
 -Wcalloc-transposed-args

Fix calloc() call against GCC 14 directive -Wcalloc-transposed-args that
makes GCC to complain with an warning/error trace message.

No functional change.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 503e14d14896b8f91c08e75ae9cdb2753850eab8 drivers: stm32_cpu_opp: fix against GCC 14 -Wcalloc-transposed-args]
Change-Id: Ic3be79ec95ce6e75bf76f66f30685d7f498b050a
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/407340
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/drivers/stm32_cpu_opp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/drivers/stm32_cpu_opp.c b/core/drivers/stm32_cpu_opp.c
index 8ff8fe39d..cf80cb195 100644
--- a/core/drivers/stm32_cpu_opp.c
+++ b/core/drivers/stm32_cpu_opp.c
@@ -671,7 +671,7 @@ static TEE_Result stm32_cpu_opp_get_dt_subnode(const void *fdt, int node)
 		if (stm32_cpu_opp_is_supported(fdt, subnode) == TEE_SUCCESS)
 			cpu_opp.opp_count++;
 
-	cpu_opp.dvfs = calloc(1, cpu_opp.opp_count * sizeof(*cpu_opp.dvfs));
+	cpu_opp.dvfs = calloc(cpu_opp.opp_count, sizeof(*cpu_opp.dvfs));
 	if (!cpu_opp.dvfs)
 		return TEE_ERROR_OUT_OF_MEMORY;
 
-- 
2.47.2

